### PR TITLE
feat: Cookbook Onboard integration

### PR DESCRIPTION
**Pull request by AE-0h**
## Preview
Preview link: https://rootstock-devportal-g9wtawg2u-cookbookdev.vercel.app/

## Ask Cookbook AI Assistant
Cookbook's Ask Cookbook AI assistant and co-pilot is trained on all existing Rootstock resources (source code, docs website, etc.) and is available as a standalone modal, embeddable as a button on any page (recommended for technical docs). It answers developer questions about building on Rootstock, acting as an enhanced and streamlined technical documentation search tool as well as a Solidity coding co-pilot.

Ask Cookbook AI can also access context from thousands of data sources indexed by Cookbook.dev in addition to Rootstock-specific data sources, providing the best blockchain developer-focused answers of any chatbot on the market. Cookbook will assist in the tuning and calibration of the AI assistant to ensure the highest answer quality.

image
  

## Cookbook Onboard Playground
The Cookbook Onboard Playground embeds a Solidity IDE with smart contract templates displayed alongside the Ask Cookbook AI Assistant.

Developers can search for and deploy Cookbook’s entire smart contract library from your technical documentation.

Features:
- Deploy smart contracts or entire protocols to Rootstock without leaving the docs
- Acts as a free, no-code or low-code token and NFT launchpad
- Save source code as a Foundry, Hardhat, or Scaffold project or export it to Remix or Chain IDE in one click

image

## Testing

* We built and deployed a staging site to test all aspects of site.  Preview is at the top of the PR for your review.

### CODE REVIEW

File: src/theme/Navbar/AskCookbook/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { Suspense } from "react";
+import BrowserOnly from "@docusaurus/BrowserOnly";
+
+/** It's a public API key, so it's safe to expose it here */
+const COOKBOOK_PUBLIC_API_KEY =
```
Comment (owans) [inline src/theme/Navbar/AskCookbook/index.js]: Hi @AE-0h According to our policy, we use Vercel Secret for API Keys. 

This Key and value has been added to Rootstock Vercel account, so we can remove from here and refer to the vercel env

https://vercel.com/docs/projects/environment-variables/sensitive-environment-variables

File: src/theme/Navbar/MobileSidebar/Search/index.js [hunk 2]
```diff
@@ -3,7 +3,6 @@ import React from 'react';
 import NavbarSearch from "@theme/Navbar/Search";
 import SearchBar from "@theme/SearchBar";
 import {useThemeConfig} from "@docusaurus/theme-common";
-import AIButton from "@theme/Navbar/AIButton";
 
 export default function NavbarMobileSidebarSearch() {
```
Comment (owans) [inline src/theme/Navbar/MobileSidebar/Search/index.js:7]: There's a slight overlap with the mobile view and the light and dark mode toggle

Screenshot 2024-11-28 at 12 40 47
Screenshot 2024-11-28 at 12 42 12

**Review by owans — COMMENTED** [review]
@AE-0h Thank you for adding the Cookbook service. Its really cool, looking forward to the embedded button and how this update will improve the experience for developers on Rootstock.

File: src/theme/Navbar/MobileSidebar/Search/index.js (hunk 2 above)
Comment (owans) [reply src/theme/Navbar/MobileSidebar/Search/index.js:7]: Hello @AE-0h Wondering if you got the chance to implement the feedback, please let me know. 

We could also handle it internally if you prefer.

Thanks

Comment (AE-0h) [reply src/theme/Navbar/MobileSidebar/Search/index.js:7]: Button placement/overlap issue with toggle is patched and was done through cms so no change to commit was required for that.  I updated the env var flow to use the docusaurus config.  I created a cookbook property and then call the value of that property in my AskCookbook component.  Please let me know if you require any further adjustments!

Comment (owans) [reply src/theme/Navbar/MobileSidebar/Search/index.js:7]: Awesome. Thank you!

**Comment by owans** [comment]
There's an error with the deployment @AE-0h

**Comment by owans** [comment]
There's an error on vercel with the deployment @AE-0h 

Screenshot 2024-12-05 at 13 05 47

**Comment by AE-0h** [comment]
> There's an error on vercel with the deployment @AE-0h
> 
> Screenshot 2024-12-05 at 13 05 47

looking into it now

**Review by owans — APPROVED** [review]
LGTM!

**Comment by owans** [comment]
Hi @AE-0h Cookbook is live on the docs 🎉! Thanks to you and your team.

QQ:

Is there a dashboard instance of the tool? To track usage? I logged in to the main account on cookbook.dev but not sure its the right one. Also please share the guide on how to use the embeddable button within the docs if there's one.

Thank you.

**Pull request merged**